src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+            if (fNewDefault) {
+               system.assertEquals(system.today(), addr.Latest_Start_Date__c);
+               system.assertEquals(null, addr.Latest_End_Date__c);
+            } else {
+               system.assertEquals(system.today(), addr.Latest_End_Date__c);                
+            }
+        }        
+    }
+
+    /*********************************************************************************************************
+    @description Update contacts' mailing address to cause creation of new default Address object.  
+    verify:
+        new Default Address created
+        Contacts get new Address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrNew() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(4);
+
+        // modify some of the Contact addresses directly
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'Direct Street Edit';
+            contactsAddrs.contacts[i].MailingCity = 'Direct City Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'another Street Edit';
+            contactsAddrs.contacts[i].MailingCity = 'another City Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'final Street Edit';
+            contactsAddrs.contacts[i].MailingCity = 'final City Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.contains('final Street Edit'));
+            system.assert(contact.MailingCity.contains('final City Edit'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+         // verify the previous addresses got Default cleared.
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.